main.go

```diff
@@ -200,12 +232,52 @@ func verticalScaling() error {
 	return nil
 }
 
-func getAWSClients() (*sqs.SQS, *rds.RDS, error) {
+func updateMemoryAlarm(client *cloudwatch.CloudWatch, alarmName, instanceClass string) error {
+	alarm, err := getExistingAlarm(client, alarmName, instanceClass)
+	if err != nil {
+		return errors.Wrap(err, "Failed to get existing Cloudwatch alarm")
+	}
+	_, err = client.PutMetricAlarm(&cloudwatch.PutMetricAlarmInput{
+		AlarmName:          aws.String(alarmName),
+		Metrics:            alarm.Metrics,
+		EvaluationPeriods:  alarm.EvaluationPeriods,
+		ComparisonOperator: alarm.ComparisonOperator,
+		Threshold:          alarm.Threshold,
+	})
+	if err != nil {
+		return errors.Wrap(err, "Failed to update Cloudwatch alarm metric")
+	}
+	return nil
+}
+
+func getExistingAlarm(client *cloudwatch.CloudWatch, alarmName, instanceClass string) (*cloudwatch.MetricAlarm, error) {
+	alarms, err := client.DescribeAlarms(&cloudwatch.DescribeAlarmsInput{
+		AlarmNames: []*string{&alarmName},
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to describe Cloudwatch alarms")
+	}
+	if len(alarms.MetricAlarms) > 0 {
+		for _, metricAlarm := range alarms.MetricAlarms {
+			if len(metricAlarm.Metrics) > 0 {
+				for index, metric := range metricAlarm.Metrics {
+					if *metric.Id == "e1" {
+						metricAlarm.Metrics[index].Expression = aws.String(fmt.Sprintf("m1 + %s*%s", memoryCacheProportion, DBInstanceClassMemory[instanceClass]))
```

Comment:
Can you give a quick explanation of what this does?

Reply:
@jwilander Yes of course. The database factory every time it creates a database it creates a cloudwatch alarm associated to each of the db instances in the cluster. Until know it was using the default metric, which was FreeMemory but this was not taking into account the cache memory. Now the database factory still creates a cloudwatch alarm but also creates a custom metric called e1, which is basically the default AWS FreeMemory (m1) plus the 75% of the total instance memory. This metric e1 is now used as criteria for the alarm. 

The vertical scaling is responsible for the instance type changes, but each instance type has its own total instance memory, so every time the vertical scaling scales up an instance type it needs to update the metric to cover this change.

Reply:
Thanks, makes sense 👍